pkg/controller/sync/sync_controller.go

```diff
@@ -132,6 +137,16 @@ type ReconcileSync struct {
 // and what is in the constraint.Spec
 func (r *ReconcileSync) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	timeStart := time.Now()
+
+	// namespace is excluded from sync
+	if r.configMatchSet != nil && len(r.configMatchSet.ExcludedNamespaces[match.Sync]) > 0 {
+		for _, ns := range r.configMatchSet.ExcludedNamespaces[match.Sync] {
+			if ns == request.Namespace {
+				return reconcile.Result{}, nil
```

Comment:
What are we doing about expectations here? Particularly if the user adds an exclude after expectations are set?

We should either mark excluded objects as observed or cancel their expectations.

@shomron what is your thought on the better approach?

Reply:
@maxsmythe  They should be functionally equivalent, but I'd lean towards CancelExpectations as conveying the actual intent, in case we add observability to the readiness trackers.

Reply:
added `t.CancelExpect(instance)` here